src/main/java/io/vertx/core/http/impl/ServerConnection.java

```diff
@@ -397,46 +397,49 @@ protected ChannelFuture sendFile(RandomAccessFile file, long offset, long length
 
   private void processMessage(Object msg) {
 
-    if (msg instanceof HttpRequest) {
-      HttpRequest request = (HttpRequest) msg;
-      DecoderResult result = ((HttpObject) msg).getDecoderResult();
+    if (msg instanceof HttpObject) {
+      HttpObject obj = (HttpObject) msg;
+      DecoderResult result = obj.decoderResult();
       if (result.isFailure()) {
         Throwable cause = result.cause();
         if (cause instanceof TooLongFrameException) {
           String causeMsg = cause.getMessage();
           HttpResponseStatus status = causeMsg.startsWith("An HTTP line is larger than") ? HttpResponseStatus.REQUEST_URI_TOO_LONG : HttpResponseStatus.BAD_REQUEST;
-          DefaultFullHttpResponse resp = new DefaultFullHttpResponse(request.protocolVersion(), status);
+          DefaultFullHttpResponse resp = new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, status);
```

Comment:
Ideally we should keep of track the protocol version number instead of forcing 1.1 here.